src/Eris/Generator/SubsetGenerator.php

```diff
@@ -0,0 +1,72 @@
+<?php
+namespace Eris\Generator;
+
+// TODO: dependency on ForAll is bad,
+// maybe inject the relative size?
+use Eris\Quantifier\ForAll;
+use Eris\Generator;
+
+/**
+ * @param array $universe
+ * @return SubsetGenerator
+ */
+function subset($input)
+{
+    return new SubsetGenerator($input);
+}
+
+class SubsetGenerator implements Generator
+{
+    private $universe;
+    
+    public function __construct(array $universe)
+    {
+        $this->universe = $universe;
+    }
+
+    public function __invoke($size)
```

Comment:
we could configure a Generator to require $size (absolute) or $relativeSize (from 0.0 to 1.0), opinions?